// @aws-sdk/client-sesv2/dist-cjs/models/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvalidNextTokenException = exports.SendingPausedException = exports.MessageRejected = exports.MailFromDomainNotVerifiedException = exports.LimitExceededException = exports.ConflictException = exports.ConcurrentModificationException = exports.TooManyRequestsException = exports.NotFoundException = exports.InternalServiceErrorException = exports.BadRequestException = exports.AlreadyExistsException = exports.AccountSuspendedException = void 0;
const SESv2ServiceException_1 = require("./SESv2ServiceException");
class AccountSuspendedException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "AccountSuspendedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccountSuspendedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccountSuspendedException.prototype);
    }
}
exports.AccountSuspendedException = AccountSuspendedException;
class AlreadyExistsException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "AlreadyExistsException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AlreadyExistsException.prototype);
    }
}
exports.AlreadyExistsException = AlreadyExistsException;
class BadRequestException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "BadRequestException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BadRequestException.prototype);
    }
}
exports.BadRequestException = BadRequestException;
class InternalServiceErrorException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "InternalServiceErrorException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalServiceErrorException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServiceErrorException.prototype);
    }
}
exports.InternalServiceErrorException = InternalServiceErrorException;
class NotFoundException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "NotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotFoundException.prototype);
    }
}
exports.NotFoundException = NotFoundException;
class TooManyRequestsException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "TooManyRequestsException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
    }
}
exports.TooManyRequestsException = TooManyRequestsException;
class ConcurrentModificationException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "ConcurrentModificationException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "ConcurrentModificationException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentModificationException.prototype);
    }
}
exports.ConcurrentModificationException = ConcurrentModificationException;
class ConflictException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
exports.ConflictException = ConflictException;
class LimitExceededException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "LimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
    }
}
exports.LimitExceededException = LimitExceededException;
class MailFromDomainNotVerifiedException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "MailFromDomainNotVerifiedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MailFromDomainNotVerifiedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MailFromDomainNotVerifiedException.prototype);
    }
}
exports.MailFromDomainNotVerifiedException = MailFromDomainNotVerifiedException;
class MessageRejected extends SESv2ServiceException_1.SESv2ServiceException {
    name = "MessageRejected";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MessageRejected",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MessageRejected.prototype);
    }
}
exports.MessageRejected = MessageRejected;
class SendingPausedException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "SendingPausedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "SendingPausedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SendingPausedException.prototype);
    }
}
exports.SendingPausedException = SendingPausedException;
class InvalidNextTokenException extends SESv2ServiceException_1.SESv2ServiceException {
    name = "InvalidNextTokenException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidNextTokenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidNextTokenException.prototype);
    }
}
exports.InvalidNextTokenException = InvalidNextTokenException;
