// @aws-sdk/client-sesv2/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var SESv2ServiceException = require('./models/SESv2ServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "ses",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class SESv2Client extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultSESv2HttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
                "aws.auth#sigv4a": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class BatchGetMetricDataCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "BatchGetMetricData", {})
    .n("SESv2Client", "BatchGetMetricDataCommand")
    .sc(schemas_0.BatchGetMetricData$)
    .build() {
}

class CancelExportJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CancelExportJob", {})
    .n("SESv2Client", "CancelExportJobCommand")
    .sc(schemas_0.CancelExportJob$)
    .build() {
}

class CreateConfigurationSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateConfigurationSet", {})
    .n("SESv2Client", "CreateConfigurationSetCommand")
    .sc(schemas_0.CreateConfigurationSet$)
    .build() {
}

class CreateConfigurationSetEventDestinationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateConfigurationSetEventDestination", {})
    .n("SESv2Client", "CreateConfigurationSetEventDestinationCommand")
    .sc(schemas_0.CreateConfigurationSetEventDestination$)
    .build() {
}

class CreateContactCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateContact", {})
    .n("SESv2Client", "CreateContactCommand")
    .sc(schemas_0.CreateContact$)
    .build() {
}

class CreateContactListCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateContactList", {})
    .n("SESv2Client", "CreateContactListCommand")
    .sc(schemas_0.CreateContactList$)
    .build() {
}

class CreateCustomVerificationEmailTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateCustomVerificationEmailTemplate", {})
    .n("SESv2Client", "CreateCustomVerificationEmailTemplateCommand")
    .sc(schemas_0.CreateCustomVerificationEmailTemplate$)
    .build() {
}

class CreateDedicatedIpPoolCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateDedicatedIpPool", {})
    .n("SESv2Client", "CreateDedicatedIpPoolCommand")
    .sc(schemas_0.CreateDedicatedIpPool$)
    .build() {
}

class CreateDeliverabilityTestReportCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateDeliverabilityTestReport", {})
    .n("SESv2Client", "CreateDeliverabilityTestReportCommand")
    .sc(schemas_0.CreateDeliverabilityTestReport$)
    .build() {
}

class CreateEmailIdentityCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateEmailIdentity", {})
    .n("SESv2Client", "CreateEmailIdentityCommand")
    .sc(schemas_0.CreateEmailIdentity$)
    .build() {
}

class CreateEmailIdentityPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateEmailIdentityPolicy", {})
    .n("SESv2Client", "CreateEmailIdentityPolicyCommand")
    .sc(schemas_0.CreateEmailIdentityPolicy$)
    .build() {
}

class CreateEmailTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateEmailTemplate", {})
    .n("SESv2Client", "CreateEmailTemplateCommand")
    .sc(schemas_0.CreateEmailTemplate$)
    .build() {
}

class CreateExportJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateExportJob", {})
    .n("SESv2Client", "CreateExportJobCommand")
    .sc(schemas_0.CreateExportJob$)
    .build() {
}

class CreateImportJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateImportJob", {})
    .n("SESv2Client", "CreateImportJobCommand")
    .sc(schemas_0.CreateImportJob$)
    .build() {
}

class CreateMultiRegionEndpointCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateMultiRegionEndpoint", {})
    .n("SESv2Client", "CreateMultiRegionEndpointCommand")
    .sc(schemas_0.CreateMultiRegionEndpoint$)
    .build() {
}

class CreateTenantCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateTenant", {})
    .n("SESv2Client", "CreateTenantCommand")
    .sc(schemas_0.CreateTenant$)
    .build() {
}

class CreateTenantResourceAssociationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "CreateTenantResourceAssociation", {})
    .n("SESv2Client", "CreateTenantResourceAssociationCommand")
    .sc(schemas_0.CreateTenantResourceAssociation$)
    .build() {
}

class DeleteConfigurationSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteConfigurationSet", {})
    .n("SESv2Client", "DeleteConfigurationSetCommand")
    .sc(schemas_0.DeleteConfigurationSet$)
    .build() {
}

class DeleteConfigurationSetEventDestinationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteConfigurationSetEventDestination", {})
    .n("SESv2Client", "DeleteConfigurationSetEventDestinationCommand")
    .sc(schemas_0.DeleteConfigurationSetEventDestination$)
    .build() {
}

class DeleteContactCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteContact", {})
    .n("SESv2Client", "DeleteContactCommand")
    .sc(schemas_0.DeleteContact$)
    .build() {
}

class DeleteContactListCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteContactList", {})
    .n("SESv2Client", "DeleteContactListCommand")
    .sc(schemas_0.DeleteContactList$)
    .build() {
}

class DeleteCustomVerificationEmailTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteCustomVerificationEmailTemplate", {})
    .n("SESv2Client", "DeleteCustomVerificationEmailTemplateCommand")
    .sc(schemas_0.DeleteCustomVerificationEmailTemplate$)
    .build() {
}

class DeleteDedicatedIpPoolCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteDedicatedIpPool", {})
    .n("SESv2Client", "DeleteDedicatedIpPoolCommand")
    .sc(schemas_0.DeleteDedicatedIpPool$)
    .build() {
}

class DeleteEmailIdentityCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteEmailIdentity", {})
    .n("SESv2Client", "DeleteEmailIdentityCommand")
    .sc(schemas_0.DeleteEmailIdentity$)
    .build() {
}

class DeleteEmailIdentityPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteEmailIdentityPolicy", {})
    .n("SESv2Client", "DeleteEmailIdentityPolicyCommand")
    .sc(schemas_0.DeleteEmailIdentityPolicy$)
    .build() {
}

class DeleteEmailTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteEmailTemplate", {})
    .n("SESv2Client", "DeleteEmailTemplateCommand")
    .sc(schemas_0.DeleteEmailTemplate$)
    .build() {
}

class DeleteMultiRegionEndpointCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteMultiRegionEndpoint", {})
    .n("SESv2Client", "DeleteMultiRegionEndpointCommand")
    .sc(schemas_0.DeleteMultiRegionEndpoint$)
    .build() {
}

class DeleteSuppressedDestinationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteSuppressedDestination", {})
    .n("SESv2Client", "DeleteSuppressedDestinationCommand")
    .sc(schemas_0.DeleteSuppressedDestination$)
    .build() {
}

class DeleteTenantCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteTenant", {})
    .n("SESv2Client", "DeleteTenantCommand")
    .sc(schemas_0.DeleteTenant$)
    .build() {
}

class DeleteTenantResourceAssociationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "DeleteTenantResourceAssociation", {})
    .n("SESv2Client", "DeleteTenantResourceAssociationCommand")
    .sc(schemas_0.DeleteTenantResourceAssociation$)
    .build() {
}

class GetAccountCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetAccount", {})
    .n("SESv2Client", "GetAccountCommand")
    .sc(schemas_0.GetAccount$)
    .build() {
}

class GetBlacklistReportsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetBlacklistReports", {})
    .n("SESv2Client", "GetBlacklistReportsCommand")
    .sc(schemas_0.GetBlacklistReports$)
    .build() {
}

class GetConfigurationSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetConfigurationSet", {})
    .n("SESv2Client", "GetConfigurationSetCommand")
    .sc(schemas_0.GetConfigurationSet$)
    .build() {
}

class GetConfigurationSetEventDestinationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetConfigurationSetEventDestinations", {})
    .n("SESv2Client", "GetConfigurationSetEventDestinationsCommand")
    .sc(schemas_0.GetConfigurationSetEventDestinations$)
    .build() {
}

class GetContactCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetContact", {})
    .n("SESv2Client", "GetContactCommand")
    .sc(schemas_0.GetContact$)
    .build() {
}

class GetContactListCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetContactList", {})
    .n("SESv2Client", "GetContactListCommand")
    .sc(schemas_0.GetContactList$)
    .build() {
}

class GetCustomVerificationEmailTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetCustomVerificationEmailTemplate", {})
    .n("SESv2Client", "GetCustomVerificationEmailTemplateCommand")
    .sc(schemas_0.GetCustomVerificationEmailTemplate$)
    .build() {
}

class GetDedicatedIpCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetDedicatedIp", {})
    .n("SESv2Client", "GetDedicatedIpCommand")
    .sc(schemas_0.GetDedicatedIp$)
    .build() {
}

class GetDedicatedIpPoolCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetDedicatedIpPool", {})
    .n("SESv2Client", "GetDedicatedIpPoolCommand")
    .sc(schemas_0.GetDedicatedIpPool$)
    .build() {
}

class GetDedicatedIpsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetDedicatedIps", {})
    .n("SESv2Client", "GetDedicatedIpsCommand")
    .sc(schemas_0.GetDedicatedIps$)
    .build() {
}

class GetDeliverabilityDashboardOptionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetDeliverabilityDashboardOptions", {})
    .n("SESv2Client", "GetDeliverabilityDashboardOptionsCommand")
    .sc(schemas_0.GetDeliverabilityDashboardOptions$)
    .build() {
}

class GetDeliverabilityTestReportCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetDeliverabilityTestReport", {})
    .n("SESv2Client", "GetDeliverabilityTestReportCommand")
    .sc(schemas_0.GetDeliverabilityTestReport$)
    .build() {
}

class GetDomainDeliverabilityCampaignCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetDomainDeliverabilityCampaign", {})
    .n("SESv2Client", "GetDomainDeliverabilityCampaignCommand")
    .sc(schemas_0.GetDomainDeliverabilityCampaign$)
    .build() {
}

class GetDomainStatisticsReportCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetDomainStatisticsReport", {})
    .n("SESv2Client", "GetDomainStatisticsReportCommand")
    .sc(schemas_0.GetDomainStatisticsReport$)
    .build() {
}

class GetEmailAddressInsightsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetEmailAddressInsights", {})
    .n("SESv2Client", "GetEmailAddressInsightsCommand")
    .sc(schemas_0.GetEmailAddressInsights$)
    .build() {
}

class GetEmailIdentityCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetEmailIdentity", {})
    .n("SESv2Client", "GetEmailIdentityCommand")
    .sc(schemas_0.GetEmailIdentity$)
    .build() {
}

class GetEmailIdentityPoliciesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetEmailIdentityPolicies", {})
    .n("SESv2Client", "GetEmailIdentityPoliciesCommand")
    .sc(schemas_0.GetEmailIdentityPolicies$)
    .build() {
}

class GetEmailTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetEmailTemplate", {})
    .n("SESv2Client", "GetEmailTemplateCommand")
    .sc(schemas_0.GetEmailTemplate$)
    .build() {
}

class GetExportJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetExportJob", {})
    .n("SESv2Client", "GetExportJobCommand")
    .sc(schemas_0.GetExportJob$)
    .build() {
}

class GetImportJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetImportJob", {})
    .n("SESv2Client", "GetImportJobCommand")
    .sc(schemas_0.GetImportJob$)
    .build() {
}

class GetMessageInsightsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetMessageInsights", {})
    .n("SESv2Client", "GetMessageInsightsCommand")
    .sc(schemas_0.GetMessageInsights$)
    .build() {
}

class GetMultiRegionEndpointCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetMultiRegionEndpoint", {})
    .n("SESv2Client", "GetMultiRegionEndpointCommand")
    .sc(schemas_0.GetMultiRegionEndpoint$)
    .build() {
}

class GetReputationEntityCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetReputationEntity", {})
    .n("SESv2Client", "GetReputationEntityCommand")
    .sc(schemas_0.GetReputationEntity$)
    .build() {
}

class GetSuppressedDestinationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetSuppressedDestination", {})
    .n("SESv2Client", "GetSuppressedDestinationCommand")
    .sc(schemas_0.GetSuppressedDestination$)
    .build() {
}

class GetTenantCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "GetTenant", {})
    .n("SESv2Client", "GetTenantCommand")
    .sc(schemas_0.GetTenant$)
    .build() {
}

class ListConfigurationSetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListConfigurationSets", {})
    .n("SESv2Client", "ListConfigurationSetsCommand")
    .sc(schemas_0.ListConfigurationSets$)
    .build() {
}

class ListContactListsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListContactLists", {})
    .n("SESv2Client", "ListContactListsCommand")
    .sc(schemas_0.ListContactLists$)
    .build() {
}

class ListContactsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListContacts", {})
    .n("SESv2Client", "ListContactsCommand")
    .sc(schemas_0.ListContacts$)
    .build() {
}

class ListCustomVerificationEmailTemplatesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListCustomVerificationEmailTemplates", {})
    .n("SESv2Client", "ListCustomVerificationEmailTemplatesCommand")
    .sc(schemas_0.ListCustomVerificationEmailTemplates$)
    .build() {
}

class ListDedicatedIpPoolsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListDedicatedIpPools", {})
    .n("SESv2Client", "ListDedicatedIpPoolsCommand")
    .sc(schemas_0.ListDedicatedIpPools$)
    .build() {
}

class ListDeliverabilityTestReportsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListDeliverabilityTestReports", {})
    .n("SESv2Client", "ListDeliverabilityTestReportsCommand")
    .sc(schemas_0.ListDeliverabilityTestReports$)
    .build() {
}

class ListDomainDeliverabilityCampaignsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListDomainDeliverabilityCampaigns", {})
    .n("SESv2Client", "ListDomainDeliverabilityCampaignsCommand")
    .sc(schemas_0.ListDomainDeliverabilityCampaigns$)
    .build() {
}

class ListEmailIdentitiesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListEmailIdentities", {})
    .n("SESv2Client", "ListEmailIdentitiesCommand")
    .sc(schemas_0.ListEmailIdentities$)
    .build() {
}

class ListEmailTemplatesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListEmailTemplates", {})
    .n("SESv2Client", "ListEmailTemplatesCommand")
    .sc(schemas_0.ListEmailTemplates$)
    .build() {
}

class ListExportJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListExportJobs", {})
    .n("SESv2Client", "ListExportJobsCommand")
    .sc(schemas_0.ListExportJobs$)
    .build() {
}

class ListImportJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListImportJobs", {})
    .n("SESv2Client", "ListImportJobsCommand")
    .sc(schemas_0.ListImportJobs$)
    .build() {
}

class ListMultiRegionEndpointsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListMultiRegionEndpoints", {})
    .n("SESv2Client", "ListMultiRegionEndpointsCommand")
    .sc(schemas_0.ListMultiRegionEndpoints$)
    .build() {
}

class ListRecommendationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListRecommendations", {})
    .n("SESv2Client", "ListRecommendationsCommand")
    .sc(schemas_0.ListRecommendations$)
    .build() {
}

class ListReputationEntitiesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListReputationEntities", {})
    .n("SESv2Client", "ListReputationEntitiesCommand")
    .sc(schemas_0.ListReputationEntities$)
    .build() {
}

class ListResourceTenantsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListResourceTenants", {})
    .n("SESv2Client", "ListResourceTenantsCommand")
    .sc(schemas_0.ListResourceTenants$)
    .build() {
}

class ListSuppressedDestinationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListSuppressedDestinations", {})
    .n("SESv2Client", "ListSuppressedDestinationsCommand")
    .sc(schemas_0.ListSuppressedDestinations$)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListTagsForResource", {})
    .n("SESv2Client", "ListTagsForResourceCommand")
    .sc(schemas_0.ListTagsForResource$)
    .build() {
}

class ListTenantResourcesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListTenantResources", {})
    .n("SESv2Client", "ListTenantResourcesCommand")
    .sc(schemas_0.ListTenantResources$)
    .build() {
}

class ListTenantsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "ListTenants", {})
    .n("SESv2Client", "ListTenantsCommand")
    .sc(schemas_0.ListTenants$)
    .build() {
}

class PutAccountDedicatedIpWarmupAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutAccountDedicatedIpWarmupAttributes", {})
    .n("SESv2Client", "PutAccountDedicatedIpWarmupAttributesCommand")
    .sc(schemas_0.PutAccountDedicatedIpWarmupAttributes$)
    .build() {
}

class PutAccountDetailsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutAccountDetails", {})
    .n("SESv2Client", "PutAccountDetailsCommand")
    .sc(schemas_0.PutAccountDetails$)
    .build() {
}

class PutAccountSendingAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutAccountSendingAttributes", {})
    .n("SESv2Client", "PutAccountSendingAttributesCommand")
    .sc(schemas_0.PutAccountSendingAttributes$)
    .build() {
}

class PutAccountSuppressionAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutAccountSuppressionAttributes", {})
    .n("SESv2Client", "PutAccountSuppressionAttributesCommand")
    .sc(schemas_0.PutAccountSuppressionAttributes$)
    .build() {
}

class PutAccountVdmAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutAccountVdmAttributes", {})
    .n("SESv2Client", "PutAccountVdmAttributesCommand")
    .sc(schemas_0.PutAccountVdmAttributes$)
    .build() {
}

class PutConfigurationSetArchivingOptionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutConfigurationSetArchivingOptions", {})
    .n("SESv2Client", "PutConfigurationSetArchivingOptionsCommand")
    .sc(schemas_0.PutConfigurationSetArchivingOptions$)
    .build() {
}

class PutConfigurationSetDeliveryOptionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutConfigurationSetDeliveryOptions", {})
    .n("SESv2Client", "PutConfigurationSetDeliveryOptionsCommand")
    .sc(schemas_0.PutConfigurationSetDeliveryOptions$)
    .build() {
}

class PutConfigurationSetReputationOptionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutConfigurationSetReputationOptions", {})
    .n("SESv2Client", "PutConfigurationSetReputationOptionsCommand")
    .sc(schemas_0.PutConfigurationSetReputationOptions$)
    .build() {
}

class PutConfigurationSetSendingOptionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutConfigurationSetSendingOptions", {})
    .n("SESv2Client", "PutConfigurationSetSendingOptionsCommand")
    .sc(schemas_0.PutConfigurationSetSendingOptions$)
    .build() {
}

class PutConfigurationSetSuppressionOptionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutConfigurationSetSuppressionOptions", {})
    .n("SESv2Client", "PutConfigurationSetSuppressionOptionsCommand")
    .sc(schemas_0.PutConfigurationSetSuppressionOptions$)
    .build() {
}

class PutConfigurationSetTrackingOptionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutConfigurationSetTrackingOptions", {})
    .n("SESv2Client", "PutConfigurationSetTrackingOptionsCommand")
    .sc(schemas_0.PutConfigurationSetTrackingOptions$)
    .build() {
}

class PutConfigurationSetVdmOptionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutConfigurationSetVdmOptions", {})
    .n("SESv2Client", "PutConfigurationSetVdmOptionsCommand")
    .sc(schemas_0.PutConfigurationSetVdmOptions$)
    .build() {
}

class PutDedicatedIpInPoolCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutDedicatedIpInPool", {})
    .n("SESv2Client", "PutDedicatedIpInPoolCommand")
    .sc(schemas_0.PutDedicatedIpInPool$)
    .build() {
}

class PutDedicatedIpPoolScalingAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutDedicatedIpPoolScalingAttributes", {})
    .n("SESv2Client", "PutDedicatedIpPoolScalingAttributesCommand")
    .sc(schemas_0.PutDedicatedIpPoolScalingAttributes$)
    .build() {
}

class PutDedicatedIpWarmupAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutDedicatedIpWarmupAttributes", {})
    .n("SESv2Client", "PutDedicatedIpWarmupAttributesCommand")
    .sc(schemas_0.PutDedicatedIpWarmupAttributes$)
    .build() {
}

class PutDeliverabilityDashboardOptionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutDeliverabilityDashboardOption", {})
    .n("SESv2Client", "PutDeliverabilityDashboardOptionCommand")
    .sc(schemas_0.PutDeliverabilityDashboardOption$)
    .build() {
}

class PutEmailIdentityConfigurationSetAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutEmailIdentityConfigurationSetAttributes", {})
    .n("SESv2Client", "PutEmailIdentityConfigurationSetAttributesCommand")
    .sc(schemas_0.PutEmailIdentityConfigurationSetAttributes$)
    .build() {
}

class PutEmailIdentityDkimAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutEmailIdentityDkimAttributes", {})
    .n("SESv2Client", "PutEmailIdentityDkimAttributesCommand")
    .sc(schemas_0.PutEmailIdentityDkimAttributes$)
    .build() {
}

class PutEmailIdentityDkimSigningAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutEmailIdentityDkimSigningAttributes", {})
    .n("SESv2Client", "PutEmailIdentityDkimSigningAttributesCommand")
    .sc(schemas_0.PutEmailIdentityDkimSigningAttributes$)
    .build() {
}

class PutEmailIdentityFeedbackAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutEmailIdentityFeedbackAttributes", {})
    .n("SESv2Client", "PutEmailIdentityFeedbackAttributesCommand")
    .sc(schemas_0.PutEmailIdentityFeedbackAttributes$)
    .build() {
}

class PutEmailIdentityMailFromAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutEmailIdentityMailFromAttributes", {})
    .n("SESv2Client", "PutEmailIdentityMailFromAttributesCommand")
    .sc(schemas_0.PutEmailIdentityMailFromAttributes$)
    .build() {
}

class PutSuppressedDestinationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "PutSuppressedDestination", {})
    .n("SESv2Client", "PutSuppressedDestinationCommand")
    .sc(schemas_0.PutSuppressedDestination$)
    .build() {
}

class SendBulkEmailCommand extends smithyClient.Command
    .classBuilder()
    .ep({
    ...commonParams,
    EndpointId: { type: "contextParams", name: "EndpointId" },
})
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "SendBulkEmail", {})
    .n("SESv2Client", "SendBulkEmailCommand")
    .sc(schemas_0.SendBulkEmail$)
    .build() {
}

class SendCustomVerificationEmailCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "SendCustomVerificationEmail", {})
    .n("SESv2Client", "SendCustomVerificationEmailCommand")
    .sc(schemas_0.SendCustomVerificationEmail$)
    .build() {
}

class SendEmailCommand extends smithyClient.Command
    .classBuilder()
    .ep({
    ...commonParams,
    EndpointId: { type: "contextParams", name: "EndpointId" },
})
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "SendEmail", {})
    .n("SESv2Client", "SendEmailCommand")
    .sc(schemas_0.SendEmail$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "TagResource", {})
    .n("SESv2Client", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class TestRenderEmailTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "TestRenderEmailTemplate", {})
    .n("SESv2Client", "TestRenderEmailTemplateCommand")
    .sc(schemas_0.TestRenderEmailTemplate$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "UntagResource", {})
    .n("SESv2Client", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateConfigurationSetEventDestinationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "UpdateConfigurationSetEventDestination", {})
    .n("SESv2Client", "UpdateConfigurationSetEventDestinationCommand")
    .sc(schemas_0.UpdateConfigurationSetEventDestination$)
    .build() {
}

class UpdateContactCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "UpdateContact", {})
    .n("SESv2Client", "UpdateContactCommand")
    .sc(schemas_0.UpdateContact$)
    .build() {
}

class UpdateContactListCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "UpdateContactList", {})
    .n("SESv2Client", "UpdateContactListCommand")
    .sc(schemas_0.UpdateContactList$)
    .build() {
}

class UpdateCustomVerificationEmailTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "UpdateCustomVerificationEmailTemplate", {})
    .n("SESv2Client", "UpdateCustomVerificationEmailTemplateCommand")
    .sc(schemas_0.UpdateCustomVerificationEmailTemplate$)
    .build() {
}

class UpdateEmailIdentityPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "UpdateEmailIdentityPolicy", {})
    .n("SESv2Client", "UpdateEmailIdentityPolicyCommand")
    .sc(schemas_0.UpdateEmailIdentityPolicy$)
    .build() {
}

class UpdateEmailTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "UpdateEmailTemplate", {})
    .n("SESv2Client", "UpdateEmailTemplateCommand")
    .sc(schemas_0.UpdateEmailTemplate$)
    .build() {
}

class UpdateReputationEntityCustomerManagedStatusCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "UpdateReputationEntityCustomerManagedStatus", {})
    .n("SESv2Client", "UpdateReputationEntityCustomerManagedStatusCommand")
    .sc(schemas_0.UpdateReputationEntityCustomerManagedStatus$)
    .build() {
}

class UpdateReputationEntityPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("SimpleEmailService_v2", "UpdateReputationEntityPolicy", {})
    .n("SESv2Client", "UpdateReputationEntityPolicyCommand")
    .sc(schemas_0.UpdateReputationEntityPolicy$)
    .build() {
}

const paginateGetDedicatedIps = core.createPaginator(SESv2Client, GetDedicatedIpsCommand, "NextToken", "NextToken", "PageSize");

const paginateListConfigurationSets = core.createPaginator(SESv2Client, ListConfigurationSetsCommand, "NextToken", "NextToken", "PageSize");

const paginateListContactLists = core.createPaginator(SESv2Client, ListContactListsCommand, "NextToken", "NextToken", "PageSize");

const paginateListContacts = core.createPaginator(SESv2Client, ListContactsCommand, "NextToken", "NextToken", "PageSize");

const paginateListCustomVerificationEmailTemplates = core.createPaginator(SESv2Client, ListCustomVerificationEmailTemplatesCommand, "NextToken", "NextToken", "PageSize");

const paginateListDedicatedIpPools = core.createPaginator(SESv2Client, ListDedicatedIpPoolsCommand, "NextToken", "NextToken", "PageSize");

const paginateListDeliverabilityTestReports = core.createPaginator(SESv2Client, ListDeliverabilityTestReportsCommand, "NextToken", "NextToken", "PageSize");

const paginateListDomainDeliverabilityCampaigns = core.createPaginator(SESv2Client, ListDomainDeliverabilityCampaignsCommand, "NextToken", "NextToken", "PageSize");

const paginateListEmailIdentities = core.createPaginator(SESv2Client, ListEmailIdentitiesCommand, "NextToken", "NextToken", "PageSize");

const paginateListEmailTemplates = core.createPaginator(SESv2Client, ListEmailTemplatesCommand, "NextToken", "NextToken", "PageSize");

const paginateListExportJobs = core.createPaginator(SESv2Client, ListExportJobsCommand, "NextToken", "NextToken", "PageSize");

const paginateListImportJobs = core.createPaginator(SESv2Client, ListImportJobsCommand, "NextToken", "NextToken", "PageSize");

const paginateListMultiRegionEndpoints = core.createPaginator(SESv2Client, ListMultiRegionEndpointsCommand, "NextToken", "NextToken", "PageSize");

const paginateListRecommendations = core.createPaginator(SESv2Client, ListRecommendationsCommand, "NextToken", "NextToken", "PageSize");

const paginateListReputationEntities = core.createPaginator(SESv2Client, ListReputationEntitiesCommand, "NextToken", "NextToken", "PageSize");

const paginateListResourceTenants = core.createPaginator(SESv2Client, ListResourceTenantsCommand, "NextToken", "NextToken", "PageSize");

const paginateListSuppressedDestinations = core.createPaginator(SESv2Client, ListSuppressedDestinationsCommand, "NextToken", "NextToken", "PageSize");

const paginateListTenantResources = core.createPaginator(SESv2Client, ListTenantResourcesCommand, "NextToken", "NextToken", "PageSize");

const paginateListTenants = core.createPaginator(SESv2Client, ListTenantsCommand, "NextToken", "NextToken", "PageSize");

const commands = {
    BatchGetMetricDataCommand,
    CancelExportJobCommand,
    CreateConfigurationSetCommand,
    CreateConfigurationSetEventDestinationCommand,
    CreateContactCommand,
    CreateContactListCommand,
    CreateCustomVerificationEmailTemplateCommand,
    CreateDedicatedIpPoolCommand,
    CreateDeliverabilityTestReportCommand,
    CreateEmailIdentityCommand,
    CreateEmailIdentityPolicyCommand,
    CreateEmailTemplateCommand,
    CreateExportJobCommand,
    CreateImportJobCommand,
    CreateMultiRegionEndpointCommand,
    CreateTenantCommand,
    CreateTenantResourceAssociationCommand,
    DeleteConfigurationSetCommand,
    DeleteConfigurationSetEventDestinationCommand,
    DeleteContactCommand,
    DeleteContactListCommand,
    DeleteCustomVerificationEmailTemplateCommand,
    DeleteDedicatedIpPoolCommand,
    DeleteEmailIdentityCommand,
    DeleteEmailIdentityPolicyCommand,
    DeleteEmailTemplateCommand,
    DeleteMultiRegionEndpointCommand,
    DeleteSuppressedDestinationCommand,
    DeleteTenantCommand,
    DeleteTenantResourceAssociationCommand,
    GetAccountCommand,
    GetBlacklistReportsCommand,
    GetConfigurationSetCommand,
    GetConfigurationSetEventDestinationsCommand,
    GetContactCommand,
    GetContactListCommand,
    GetCustomVerificationEmailTemplateCommand,
    GetDedicatedIpCommand,
    GetDedicatedIpPoolCommand,
    GetDedicatedIpsCommand,
    GetDeliverabilityDashboardOptionsCommand,
    GetDeliverabilityTestReportCommand,
    GetDomainDeliverabilityCampaignCommand,
    GetDomainStatisticsReportCommand,
    GetEmailAddressInsightsCommand,
    GetEmailIdentityCommand,
    GetEmailIdentityPoliciesCommand,
    GetEmailTemplateCommand,
    GetExportJobCommand,
    GetImportJobCommand,
    GetMessageInsightsCommand,
    GetMultiRegionEndpointCommand,
    GetReputationEntityCommand,
    GetSuppressedDestinationCommand,
    GetTenantCommand,
    ListConfigurationSetsCommand,
    ListContactListsCommand,
    ListContactsCommand,
    ListCustomVerificationEmailTemplatesCommand,
    ListDedicatedIpPoolsCommand,
    ListDeliverabilityTestReportsCommand,
    ListDomainDeliverabilityCampaignsCommand,
    ListEmailIdentitiesCommand,
    ListEmailTemplatesCommand,
    ListExportJobsCommand,
    ListImportJobsCommand,
    ListMultiRegionEndpointsCommand,
    ListRecommendationsCommand,
    ListReputationEntitiesCommand,
    ListResourceTenantsCommand,
    ListSuppressedDestinationsCommand,
    ListTagsForResourceCommand,
    ListTenantResourcesCommand,
    ListTenantsCommand,
    PutAccountDedicatedIpWarmupAttributesCommand,
    PutAccountDetailsCommand,
    PutAccountSendingAttributesCommand,
    PutAccountSuppressionAttributesCommand,
    PutAccountVdmAttributesCommand,
    PutConfigurationSetArchivingOptionsCommand,
    PutConfigurationSetDeliveryOptionsCommand,
    PutConfigurationSetReputationOptionsCommand,
    PutConfigurationSetSendingOptionsCommand,
    PutConfigurationSetSuppressionOptionsCommand,
    PutConfigurationSetTrackingOptionsCommand,
    PutConfigurationSetVdmOptionsCommand,
    PutDedicatedIpInPoolCommand,
    PutDedicatedIpPoolScalingAttributesCommand,
    PutDedicatedIpWarmupAttributesCommand,
    PutDeliverabilityDashboardOptionCommand,
    PutEmailIdentityConfigurationSetAttributesCommand,
    PutEmailIdentityDkimAttributesCommand,
    PutEmailIdentityDkimSigningAttributesCommand,
    PutEmailIdentityFeedbackAttributesCommand,
    PutEmailIdentityMailFromAttributesCommand,
    PutSuppressedDestinationCommand,
    SendBulkEmailCommand,
    SendCustomVerificationEmailCommand,
    SendEmailCommand,
    TagResourceCommand,
    TestRenderEmailTemplateCommand,
    UntagResourceCommand,
    UpdateConfigurationSetEventDestinationCommand,
    UpdateContactCommand,
    UpdateContactListCommand,
    UpdateCustomVerificationEmailTemplateCommand,
    UpdateEmailIdentityPolicyCommand,
    UpdateEmailTemplateCommand,
    UpdateReputationEntityCustomerManagedStatusCommand,
    UpdateReputationEntityPolicyCommand,
};
const paginators = {
    paginateGetDedicatedIps,
    paginateListConfigurationSets,
    paginateListContactLists,
    paginateListContacts,
    paginateListCustomVerificationEmailTemplates,
    paginateListDedicatedIpPools,
    paginateListDeliverabilityTestReports,
    paginateListDomainDeliverabilityCampaigns,
    paginateListEmailIdentities,
    paginateListEmailTemplates,
    paginateListExportJobs,
    paginateListImportJobs,
    paginateListMultiRegionEndpoints,
    paginateListRecommendations,
    paginateListReputationEntities,
    paginateListResourceTenants,
    paginateListSuppressedDestinations,
    paginateListTenantResources,
    paginateListTenants,
};
class SESv2 extends SESv2Client {
}
smithyClient.createAggregatedClient(commands, SESv2, { paginators });

const ContactLanguage = {
    EN: "EN",
    JA: "JA",
};
const MailType = {
    MARKETING: "MARKETING",
    TRANSACTIONAL: "TRANSACTIONAL",
};
const ReviewStatus = {
    DENIED: "DENIED",
    FAILED: "FAILED",
    GRANTED: "GRANTED",
    PENDING: "PENDING",
};
const AttachmentContentDisposition = {
    ATTACHMENT: "ATTACHMENT",
    INLINE: "INLINE",
};
const AttachmentContentTransferEncoding = {
    BASE64: "BASE64",
    QUOTED_PRINTABLE: "QUOTED_PRINTABLE",
    SEVEN_BIT: "SEVEN_BIT",
};
const MetricDimensionName = {
    CONFIGURATION_SET: "CONFIGURATION_SET",
    EMAIL_IDENTITY: "EMAIL_IDENTITY",
    ISP: "ISP",
};
const Metric = {
    CLICK: "CLICK",
    COMPLAINT: "COMPLAINT",
    DELIVERY: "DELIVERY",
    DELIVERY_CLICK: "DELIVERY_CLICK",
    DELIVERY_COMPLAINT: "DELIVERY_COMPLAINT",
    DELIVERY_OPEN: "DELIVERY_OPEN",
    OPEN: "OPEN",
    PERMANENT_BOUNCE: "PERMANENT_BOUNCE",
    SEND: "SEND",
    TRANSIENT_BOUNCE: "TRANSIENT_BOUNCE",
};
const MetricNamespace = {
    VDM: "VDM",
};
const QueryErrorCode = {
    ACCESS_DENIED: "ACCESS_DENIED",
    INTERNAL_FAILURE: "INTERNAL_FAILURE",
};
const BehaviorOnMxFailure = {
    REJECT_MESSAGE: "REJECT_MESSAGE",
    USE_DEFAULT_VALUE: "USE_DEFAULT_VALUE",
};
const BounceType = {
    PERMANENT: "PERMANENT",
    TRANSIENT: "TRANSIENT",
    UNDETERMINED: "UNDETERMINED",
};
const BulkEmailStatus = {
    ACCOUNT_DAILY_QUOTA_EXCEEDED: "ACCOUNT_DAILY_QUOTA_EXCEEDED",
    ACCOUNT_SENDING_PAUSED: "ACCOUNT_SENDING_PAUSED",
    ACCOUNT_SUSPENDED: "ACCOUNT_SUSPENDED",
    ACCOUNT_THROTTLED: "ACCOUNT_THROTTLED",
    CONFIGURATION_SET_NOT_FOUND: "CONFIGURATION_SET_NOT_FOUND",
    CONFIGURATION_SET_SENDING_PAUSED: "CONFIGURATION_SET_SENDING_PAUSED",
    FAILED: "FAILED",
    INVALID_PARAMETER: "INVALID_PARAMETER",
    INVALID_SENDING_POOL_NAME: "INVALID_SENDING_POOL_NAME",
    MAIL_FROM_DOMAIN_NOT_VERIFIED: "MAIL_FROM_DOMAIN_NOT_VERIFIED",
    MESSAGE_REJECTED: "MESSAGE_REJECTED",
    SUCCESS: "SUCCESS",
    TEMPLATE_NOT_FOUND: "TEMPLATE_NOT_FOUND",
    TRANSIENT_FAILURE: "TRANSIENT_FAILURE",
};
const DimensionValueSource = {
    EMAIL_HEADER: "EMAIL_HEADER",
    LINK_TAG: "LINK_TAG",
    MESSAGE_TAG: "MESSAGE_TAG",
};
const SubscriptionStatus = {
    OPT_IN: "OPT_IN",
    OPT_OUT: "OPT_OUT",
};
const ContactListImportAction = {
    DELETE: "DELETE",
    PUT: "PUT",
};
const TlsPolicy = {
    OPTIONAL: "OPTIONAL",
    REQUIRE: "REQUIRE",
};
const SuppressionListReason = {
    BOUNCE: "BOUNCE",
    COMPLAINT: "COMPLAINT",
};
const FeatureStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const SuppressionConfidenceVerdictThreshold = {
    HIGH: "HIGH",
    MANAGED: "MANAGED",
    MEDIUM: "MEDIUM",
};
const HttpsPolicy = {
    OPTIONAL: "OPTIONAL",
    REQUIRE: "REQUIRE",
    REQUIRE_OPEN_ONLY: "REQUIRE_OPEN_ONLY",
};
const EventType = {
    BOUNCE: "BOUNCE",
    CLICK: "CLICK",
    COMPLAINT: "COMPLAINT",
    DELIVERY: "DELIVERY",
    DELIVERY_DELAY: "DELIVERY_DELAY",
    OPEN: "OPEN",
    REJECT: "REJECT",
    RENDERING_FAILURE: "RENDERING_FAILURE",
    SEND: "SEND",
    SUBSCRIPTION: "SUBSCRIPTION",
};
const ScalingMode = {
    MANAGED: "MANAGED",
    STANDARD: "STANDARD",
};
const DeliverabilityTestStatus = {
    COMPLETED: "COMPLETED",
    IN_PROGRESS: "IN_PROGRESS",
};
const DkimSigningAttributesOrigin = {
    AWS_SES: "AWS_SES",
    AWS_SES_AF_SOUTH_1: "AWS_SES_AF_SOUTH_1",
    AWS_SES_AP_NORTHEAST_1: "AWS_SES_AP_NORTHEAST_1",
    AWS_SES_AP_NORTHEAST_2: "AWS_SES_AP_NORTHEAST_2",
    AWS_SES_AP_NORTHEAST_3: "AWS_SES_AP_NORTHEAST_3",
    AWS_SES_AP_SOUTHEAST_1: "AWS_SES_AP_SOUTHEAST_1",
    AWS_SES_AP_SOUTHEAST_2: "AWS_SES_AP_SOUTHEAST_2",
    AWS_SES_AP_SOUTHEAST_3: "AWS_SES_AP_SOUTHEAST_3",
    AWS_SES_AP_SOUTHEAST_5: "AWS_SES_AP_SOUTHEAST_5",
    AWS_SES_AP_SOUTH_1: "AWS_SES_AP_SOUTH_1",
    AWS_SES_AP_SOUTH_2: "AWS_SES_AP_SOUTH_2",
    AWS_SES_CA_CENTRAL_1: "AWS_SES_CA_CENTRAL_1",
    AWS_SES_CA_WEST_1: "AWS_SES_CA_WEST_1",
    AWS_SES_EU_CENTRAL_1: "AWS_SES_EU_CENTRAL_1",
    AWS_SES_EU_CENTRAL_2: "AWS_SES_EU_CENTRAL_2",
    AWS_SES_EU_NORTH_1: "AWS_SES_EU_NORTH_1",
    AWS_SES_EU_SOUTH_1: "AWS_SES_EU_SOUTH_1",
    AWS_SES_EU_WEST_1: "AWS_SES_EU_WEST_1",
    AWS_SES_EU_WEST_2: "AWS_SES_EU_WEST_2",
    AWS_SES_EU_WEST_3: "AWS_SES_EU_WEST_3",
    AWS_SES_IL_CENTRAL_1: "AWS_SES_IL_CENTRAL_1",
    AWS_SES_ME_CENTRAL_1: "AWS_SES_ME_CENTRAL_1",
    AWS_SES_ME_SOUTH_1: "AWS_SES_ME_SOUTH_1",
    AWS_SES_SA_EAST_1: "AWS_SES_SA_EAST_1",
    AWS_SES_US_EAST_1: "AWS_SES_US_EAST_1",
    AWS_SES_US_EAST_2: "AWS_SES_US_EAST_2",
    AWS_SES_US_WEST_1: "AWS_SES_US_WEST_1",
    AWS_SES_US_WEST_2: "AWS_SES_US_WEST_2",
    EXTERNAL: "EXTERNAL",
};
const DkimSigningKeyLength = {
    RSA_1024_BIT: "RSA_1024_BIT",
    RSA_2048_BIT: "RSA_2048_BIT",
};
const DkimStatus = {
    FAILED: "FAILED",
    NOT_STARTED: "NOT_STARTED",
    PENDING: "PENDING",
    SUCCESS: "SUCCESS",
    TEMPORARY_FAILURE: "TEMPORARY_FAILURE",
};
const IdentityType = {
    DOMAIN: "DOMAIN",
    EMAIL_ADDRESS: "EMAIL_ADDRESS",
    MANAGED_DOMAIN: "MANAGED_DOMAIN",
};
const DeliveryEventType = {
    COMPLAINT: "COMPLAINT",
    DELIVERY: "DELIVERY",
    PERMANENT_BOUNCE: "PERMANENT_BOUNCE",
    SEND: "SEND",
    TRANSIENT_BOUNCE: "TRANSIENT_BOUNCE",
    UNDETERMINED_BOUNCE: "UNDETERMINED_BOUNCE",
};
const EngagementEventType = {
    CLICK: "CLICK",
    OPEN: "OPEN",
};
const MetricAggregation = {
    RATE: "RATE",
    VOLUME: "VOLUME",
};
const DataFormat = {
    CSV: "CSV",
    JSON: "JSON",
};
const SuppressionListImportAction = {
    DELETE: "DELETE",
    PUT: "PUT",
};
const Status = {
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
    READY: "READY",
};
const SendingStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    REINSTATED: "REINSTATED",
};
const WarmupStatus = {
    DONE: "DONE",
    IN_PROGRESS: "IN_PROGRESS",
    NOT_APPLICABLE: "NOT_APPLICABLE",
};
const DeliverabilityDashboardAccountStatus = {
    ACTIVE: "ACTIVE",
    DISABLED: "DISABLED",
    PENDING_EXPIRATION: "PENDING_EXPIRATION",
};
const EmailAddressInsightsConfidenceVerdict = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
};
const ExportSourceType = {
    MESSAGE_INSIGHTS: "MESSAGE_INSIGHTS",
    METRICS_DATA: "METRICS_DATA",
};
const JobStatus = {
    CANCELLED: "CANCELLED",
    COMPLETED: "COMPLETED",
    CREATED: "CREATED",
    FAILED: "FAILED",
    PROCESSING: "PROCESSING",
};
const MailFromDomainStatus = {
    FAILED: "FAILED",
    PENDING: "PENDING",
    SUCCESS: "SUCCESS",
    TEMPORARY_FAILURE: "TEMPORARY_FAILURE",
};
const VerificationError = {
    DNS_SERVER_ERROR: "DNS_SERVER_ERROR",
    HOST_NOT_FOUND: "HOST_NOT_FOUND",
    INVALID_VALUE: "INVALID_VALUE",
    REPLICATION_ACCESS_DENIED: "REPLICATION_ACCESS_DENIED",
    REPLICATION_PRIMARY_BYO_DKIM_NOT_SUPPORTED: "REPLICATION_PRIMARY_BYO_DKIM_NOT_SUPPORTED",
    REPLICATION_PRIMARY_INVALID_REGION: "REPLICATION_PRIMARY_INVALID_REGION",
    REPLICATION_PRIMARY_NOT_FOUND: "REPLICATION_PRIMARY_NOT_FOUND",
    REPLICATION_REPLICA_AS_PRIMARY_NOT_SUPPORTED: "REPLICATION_REPLICA_AS_PRIMARY_NOT_SUPPORTED",
    SERVICE_ERROR: "SERVICE_ERROR",
    TYPE_NOT_FOUND: "TYPE_NOT_FOUND",
};
const VerificationStatus = {
    FAILED: "FAILED",
    NOT_STARTED: "NOT_STARTED",
    PENDING: "PENDING",
    SUCCESS: "SUCCESS",
    TEMPORARY_FAILURE: "TEMPORARY_FAILURE",
};
const ReputationEntityType = {
    RESOURCE: "RESOURCE",
};
const RecommendationImpact = {
    HIGH: "HIGH",
    LOW: "LOW",
};
const ImportDestinationType = {
    CONTACT_LIST: "CONTACT_LIST",
    SUPPRESSION_LIST: "SUPPRESSION_LIST",
};
const ListRecommendationsFilterKey = {
    IMPACT: "IMPACT",
    RESOURCE_ARN: "RESOURCE_ARN",
    STATUS: "STATUS",
    TYPE: "TYPE",
};
const RecommendationStatus = {
    FIXED: "FIXED",
    OPEN: "OPEN",
};
const RecommendationType = {
    BIMI: "BIMI",
    BOUNCE: "BOUNCE",
    COMPLAINT: "COMPLAINT",
    DKIM: "DKIM",
    DMARC: "DMARC",
    FEEDBACK_3P: "FEEDBACK_3P",
    IP_LISTING: "IP_LISTING",
    SPF: "SPF",
};
const ReputationEntityFilterKey = {
    ENTITY_REFERENCE_PREFIX: "ENTITY_REFERENCE_PREFIX",
    ENTITY_TYPE: "ENTITY_TYPE",
    REPUTATION_IMPACT: "REPUTATION_IMPACT",
    STATUS: "SENDING_STATUS",
};
const ListTenantResourcesFilterKey = {
    RESOURCE_TYPE: "RESOURCE_TYPE",
};
const ResourceType = {
    CONFIGURATION_SET: "CONFIGURATION_SET",
    EMAIL_IDENTITY: "EMAIL_IDENTITY",
    EMAIL_TEMPLATE: "EMAIL_TEMPLATE",
};

exports.$Command = smithyClient.Command;
exports.__Client = smithyClient.Client;
exports.SESv2ServiceException = SESv2ServiceException.SESv2ServiceException;
exports.AttachmentContentDisposition = AttachmentContentDisposition;
exports.AttachmentContentTransferEncoding = AttachmentContentTransferEncoding;
exports.BatchGetMetricDataCommand = BatchGetMetricDataCommand;
exports.BehaviorOnMxFailure = BehaviorOnMxFailure;
exports.BounceType = BounceType;
exports.BulkEmailStatus = BulkEmailStatus;
exports.CancelExportJobCommand = CancelExportJobCommand;
exports.ContactLanguage = ContactLanguage;
exports.ContactListImportAction = ContactListImportAction;
exports.CreateConfigurationSetCommand = CreateConfigurationSetCommand;
exports.CreateConfigurationSetEventDestinationCommand = CreateConfigurationSetEventDestinationCommand;
exports.CreateContactCommand = CreateContactCommand;
exports.CreateContactListCommand = CreateContactListCommand;
exports.CreateCustomVerificationEmailTemplateCommand = CreateCustomVerificationEmailTemplateCommand;
exports.CreateDedicatedIpPoolCommand = CreateDedicatedIpPoolCommand;
exports.CreateDeliverabilityTestReportCommand = CreateDeliverabilityTestReportCommand;
exports.CreateEmailIdentityCommand = CreateEmailIdentityCommand;
exports.CreateEmailIdentityPolicyCommand = CreateEmailIdentityPolicyCommand;
exports.CreateEmailTemplateCommand = CreateEmailTemplateCommand;
exports.CreateExportJobCommand = CreateExportJobCommand;
exports.CreateImportJobCommand = CreateImportJobCommand;
exports.CreateMultiRegionEndpointCommand = CreateMultiRegionEndpointCommand;
exports.CreateTenantCommand = CreateTenantCommand;
exports.CreateTenantResourceAssociationCommand = CreateTenantResourceAssociationCommand;
exports.DataFormat = DataFormat;
exports.DeleteConfigurationSetCommand = DeleteConfigurationSetCommand;
exports.DeleteConfigurationSetEventDestinationCommand = DeleteConfigurationSetEventDestinationCommand;
exports.DeleteContactCommand = DeleteContactCommand;
exports.DeleteContactListCommand = DeleteContactListCommand;
exports.DeleteCustomVerificationEmailTemplateCommand = DeleteCustomVerificationEmailTemplateCommand;
exports.DeleteDedicatedIpPoolCommand = DeleteDedicatedIpPoolCommand;
exports.DeleteEmailIdentityCommand = DeleteEmailIdentityCommand;
exports.DeleteEmailIdentityPolicyCommand = DeleteEmailIdentityPolicyCommand;
exports.DeleteEmailTemplateCommand = DeleteEmailTemplateCommand;
exports.DeleteMultiRegionEndpointCommand = DeleteMultiRegionEndpointCommand;
exports.DeleteSuppressedDestinationCommand = DeleteSuppressedDestinationCommand;
exports.DeleteTenantCommand = DeleteTenantCommand;
exports.DeleteTenantResourceAssociationCommand = DeleteTenantResourceAssociationCommand;
exports.DeliverabilityDashboardAccountStatus = DeliverabilityDashboardAccountStatus;
exports.DeliverabilityTestStatus = DeliverabilityTestStatus;
exports.DeliveryEventType = DeliveryEventType;
exports.DimensionValueSource = DimensionValueSource;
exports.DkimSigningAttributesOrigin = DkimSigningAttributesOrigin;
exports.DkimSigningKeyLength = DkimSigningKeyLength;
exports.DkimStatus = DkimStatus;
exports.EmailAddressInsightsConfidenceVerdict = EmailAddressInsightsConfidenceVerdict;
exports.EngagementEventType = EngagementEventType;
exports.EventType = EventType;
exports.ExportSourceType = ExportSourceType;
exports.FeatureStatus = FeatureStatus;
exports.GetAccountCommand = GetAccountCommand;
exports.GetBlacklistReportsCommand = GetBlacklistReportsCommand;
exports.GetConfigurationSetCommand = GetConfigurationSetCommand;
exports.GetConfigurationSetEventDestinationsCommand = GetConfigurationSetEventDestinationsCommand;
exports.GetContactCommand = GetContactCommand;
exports.GetContactListCommand = GetContactListCommand;
exports.GetCustomVerificationEmailTemplateCommand = GetCustomVerificationEmailTemplateCommand;
exports.GetDedicatedIpCommand = GetDedicatedIpCommand;
exports.GetDedicatedIpPoolCommand = GetDedicatedIpPoolCommand;
exports.GetDedicatedIpsCommand = GetDedicatedIpsCommand;
exports.GetDeliverabilityDashboardOptionsCommand = GetDeliverabilityDashboardOptionsCommand;
exports.GetDeliverabilityTestReportCommand = GetDeliverabilityTestReportCommand;
exports.GetDomainDeliverabilityCampaignCommand = GetDomainDeliverabilityCampaignCommand;
exports.GetDomainStatisticsReportCommand = GetDomainStatisticsReportCommand;
exports.GetEmailAddressInsightsCommand = GetEmailAddressInsightsCommand;
exports.GetEmailIdentityCommand = GetEmailIdentityCommand;
exports.GetEmailIdentityPoliciesCommand = GetEmailIdentityPoliciesCommand;
exports.GetEmailTemplateCommand = GetEmailTemplateCommand;
exports.GetExportJobCommand = GetExportJobCommand;
exports.GetImportJobCommand = GetImportJobCommand;
exports.GetMessageInsightsCommand = GetMessageInsightsCommand;
exports.GetMultiRegionEndpointCommand = GetMultiRegionEndpointCommand;
exports.GetReputationEntityCommand = GetReputationEntityCommand;
exports.GetSuppressedDestinationCommand = GetSuppressedDestinationCommand;
exports.GetTenantCommand = GetTenantCommand;
exports.HttpsPolicy = HttpsPolicy;
exports.IdentityType = IdentityType;
exports.ImportDestinationType = ImportDestinationType;
exports.JobStatus = JobStatus;
exports.ListConfigurationSetsCommand = ListConfigurationSetsCommand;
exports.ListContactListsCommand = ListContactListsCommand;
exports.ListContactsCommand = ListContactsCommand;
exports.ListCustomVerificationEmailTemplatesCommand = ListCustomVerificationEmailTemplatesCommand;
exports.ListDedicatedIpPoolsCommand = ListDedicatedIpPoolsCommand;
exports.ListDeliverabilityTestReportsCommand = ListDeliverabilityTestReportsCommand;
exports.ListDomainDeliverabilityCampaignsCommand = ListDomainDeliverabilityCampaignsCommand;
exports.ListEmailIdentitiesCommand = ListEmailIdentitiesCommand;
exports.ListEmailTemplatesCommand = ListEmailTemplatesCommand;
exports.ListExportJobsCommand = ListExportJobsCommand;
exports.ListImportJobsCommand = ListImportJobsCommand;
exports.ListMultiRegionEndpointsCommand = ListMultiRegionEndpointsCommand;
exports.ListRecommendationsCommand = ListRecommendationsCommand;
exports.ListRecommendationsFilterKey = ListRecommendationsFilterKey;
exports.ListReputationEntitiesCommand = ListReputationEntitiesCommand;
exports.ListResourceTenantsCommand = ListResourceTenantsCommand;
exports.ListSuppressedDestinationsCommand = ListSuppressedDestinationsCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTenantResourcesCommand = ListTenantResourcesCommand;
exports.ListTenantResourcesFilterKey = ListTenantResourcesFilterKey;
exports.ListTenantsCommand = ListTenantsCommand;
exports.MailFromDomainStatus = MailFromDomainStatus;
exports.MailType = MailType;
exports.Metric = Metric;
exports.MetricAggregation = MetricAggregation;
exports.MetricDimensionName = MetricDimensionName;
exports.MetricNamespace = MetricNamespace;
exports.PutAccountDedicatedIpWarmupAttributesCommand = PutAccountDedicatedIpWarmupAttributesCommand;
exports.PutAccountDetailsCommand = PutAccountDetailsCommand;
exports.PutAccountSendingAttributesCommand = PutAccountSendingAttributesCommand;
exports.PutAccountSuppressionAttributesCommand = PutAccountSuppressionAttributesCommand;
exports.PutAccountVdmAttributesCommand = PutAccountVdmAttributesCommand;
exports.PutConfigurationSetArchivingOptionsCommand = PutConfigurationSetArchivingOptionsCommand;
exports.PutConfigurationSetDeliveryOptionsCommand = PutConfigurationSetDeliveryOptionsCommand;
exports.PutConfigurationSetReputationOptionsCommand = PutConfigurationSetReputationOptionsCommand;
exports.PutConfigurationSetSendingOptionsCommand = PutConfigurationSetSendingOptionsCommand;
exports.PutConfigurationSetSuppressionOptionsCommand = PutConfigurationSetSuppressionOptionsCommand;
exports.PutConfigurationSetTrackingOptionsCommand = PutConfigurationSetTrackingOptionsCommand;
exports.PutConfigurationSetVdmOptionsCommand = PutConfigurationSetVdmOptionsCommand;
exports.PutDedicatedIpInPoolCommand = PutDedicatedIpInPoolCommand;
exports.PutDedicatedIpPoolScalingAttributesCommand = PutDedicatedIpPoolScalingAttributesCommand;
exports.PutDedicatedIpWarmupAttributesCommand = PutDedicatedIpWarmupAttributesCommand;
exports.PutDeliverabilityDashboardOptionCommand = PutDeliverabilityDashboardOptionCommand;
exports.PutEmailIdentityConfigurationSetAttributesCommand = PutEmailIdentityConfigurationSetAttributesCommand;
exports.PutEmailIdentityDkimAttributesCommand = PutEmailIdentityDkimAttributesCommand;
exports.PutEmailIdentityDkimSigningAttributesCommand = PutEmailIdentityDkimSigningAttributesCommand;
exports.PutEmailIdentityFeedbackAttributesCommand = PutEmailIdentityFeedbackAttributesCommand;
exports.PutEmailIdentityMailFromAttributesCommand = PutEmailIdentityMailFromAttributesCommand;
exports.PutSuppressedDestinationCommand = PutSuppressedDestinationCommand;
exports.QueryErrorCode = QueryErrorCode;
exports.RecommendationImpact = RecommendationImpact;
exports.RecommendationStatus = RecommendationStatus;
exports.RecommendationType = RecommendationType;
exports.ReputationEntityFilterKey = ReputationEntityFilterKey;
exports.ReputationEntityType = ReputationEntityType;
exports.ResourceType = ResourceType;
exports.ReviewStatus = ReviewStatus;
exports.SESv2 = SESv2;
exports.SESv2Client = SESv2Client;
exports.ScalingMode = ScalingMode;
exports.SendBulkEmailCommand = SendBulkEmailCommand;
exports.SendCustomVerificationEmailCommand = SendCustomVerificationEmailCommand;
exports.SendEmailCommand = SendEmailCommand;
exports.SendingStatus = SendingStatus;
exports.Status = Status;
exports.SubscriptionStatus = SubscriptionStatus;
exports.SuppressionConfidenceVerdictThreshold = SuppressionConfidenceVerdictThreshold;
exports.SuppressionListImportAction = SuppressionListImportAction;
exports.SuppressionListReason = SuppressionListReason;
exports.TagResourceCommand = TagResourceCommand;
exports.TestRenderEmailTemplateCommand = TestRenderEmailTemplateCommand;
exports.TlsPolicy = TlsPolicy;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateConfigurationSetEventDestinationCommand = UpdateConfigurationSetEventDestinationCommand;
exports.UpdateContactCommand = UpdateContactCommand;
exports.UpdateContactListCommand = UpdateContactListCommand;
exports.UpdateCustomVerificationEmailTemplateCommand = UpdateCustomVerificationEmailTemplateCommand;
exports.UpdateEmailIdentityPolicyCommand = UpdateEmailIdentityPolicyCommand;
exports.UpdateEmailTemplateCommand = UpdateEmailTemplateCommand;
exports.UpdateReputationEntityCustomerManagedStatusCommand = UpdateReputationEntityCustomerManagedStatusCommand;
exports.UpdateReputationEntityPolicyCommand = UpdateReputationEntityPolicyCommand;
exports.VerificationError = VerificationError;
exports.VerificationStatus = VerificationStatus;
exports.WarmupStatus = WarmupStatus;
exports.paginateGetDedicatedIps = paginateGetDedicatedIps;
exports.paginateListConfigurationSets = paginateListConfigurationSets;
exports.paginateListContactLists = paginateListContactLists;
exports.paginateListContacts = paginateListContacts;
exports.paginateListCustomVerificationEmailTemplates = paginateListCustomVerificationEmailTemplates;
exports.paginateListDedicatedIpPools = paginateListDedicatedIpPools;
exports.paginateListDeliverabilityTestReports = paginateListDeliverabilityTestReports;
exports.paginateListDomainDeliverabilityCampaigns = paginateListDomainDeliverabilityCampaigns;
exports.paginateListEmailIdentities = paginateListEmailIdentities;
exports.paginateListEmailTemplates = paginateListEmailTemplates;
exports.paginateListExportJobs = paginateListExportJobs;
exports.paginateListImportJobs = paginateListImportJobs;
exports.paginateListMultiRegionEndpoints = paginateListMultiRegionEndpoints;
exports.paginateListRecommendations = paginateListRecommendations;
exports.paginateListReputationEntities = paginateListReputationEntities;
exports.paginateListResourceTenants = paginateListResourceTenants;
exports.paginateListSuppressedDestinations = paginateListSuppressedDestinations;
exports.paginateListTenantResources = paginateListTenantResources;
exports.paginateListTenants = paginateListTenants;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
